cypress/e2e/gui/engage-sphere.spec.cy.js

```diff
@@ -0,0 +1,378 @@
+            cy.wait('@getHRCustomers');
+            cy.get('tbody tr').should('have.length', 4);
+        });
+
+        it('filters the customers by finance industry', () => {
+            cy.intercept('GET', `${CUSTOMERS_API_URL}?page=1&limit=10&size=All&industry=Finance`,
+                { fixture: 'industry/finance' }
+            ).as('getFinanceCustomers');
+            cy.get('[data-testid="industry-filter"]').select('Finance');
+            cy.wait('@getFinanceCustomers');
+            cy.get('tbody tr').should('have.length', 6);
+        });
+
+        it('filters the customers by all industries', () => {
+            cy.get('[data-testid="industry-filter"]').select('Technology');
+            cy.intercept('GET', `${CUSTOMERS_API_URL}?page=1&limit=10&size=All&industry=All`,
+                { fixture: 'all' }
+            ).as('getAllCustomers');
+            cy.get('[data-testid="industry-filter"]').select('All');
+            cy.wait('@getAllCustomers');
+            cy.get('tbody tr').should('have.length', 9);
+        });
+    });
+
+    context('View', () => {
+        it('goes back to the customers list when clicking the "Back" button', () => {
+            cy.contains('button', 'View').eq(0).click();
+            cy.contains('button', 'Back').click();
+            cy.contains('p', 'Below is our customer list.').should('be.visible');
+        });
+
+        it('shows a Loading... fallback element before the initial customers fetch', () => {
+            cy.reload();
+            cy.contains('p', 'Loading...')
+                .should('be.visible');
+        });
+    });
+
+
+    context('Acessibility', () => {
+        beforeEach(() => {
+            cy.injectAxe();
+        });
+
+        context('Light Mode', () => {
+            it('Customer table should have no accessibility issues', () => {
+                cy.checkA11y();
+            });
+
+            it('Customer details and address view should have no accessibility issues', () => {
+                cy.get('button').contains('View').click();
+                cy.checkA11y();
+            });
+
+            it('Messenger contact form should have no accessibility issues', () => {
+                //fiquei em dúvida, se há necessidade de ajustar o component Messenger Contact Form ou easter egg
+                cy.getByClassThatStartsWith('Messenger_openCloseButton').click();
+                cy.checkA11y();
+            });
+        });
+
+        context('Dark Mode', () => {
+            beforeEach(() => {
+                cy.getByClassThatStartsWith('ThemeToggle_button').click();
+                cy.get('[data-theme="dark"]').should('exist');
+            });
+
+            it('Customer table should have no accessibility issues in dark mode', () => {
+                cy.checkA11y();
+            });
+
+            it('Customer details and address view should have no accessibility issues in dark mode', () => {
+                cy.get('button').contains('View').click();
```

Comment:
```suggestion
                cy.contains('button', 'View').click();
```

Para aprender mais, leia [este documento](https://gist.github.com/wlsf82/f815a14072f9edd80aac175b12335a36#cycontains).